File: TokApp/ios/ShareExtensionBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SharedUserDefaults, NSObject)

RCT_EXTERN_METHOD(setSessionToken:(NSString *)token
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getSessionToken:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(removeSessionToken:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: TokApp/ios/lai/ShareExtensionBridge.swift
import Foundation
import React

@objc(ShareExtensionBridge)
class ShareExtensionBridge: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func getSharedData(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    
    // Access the shared UserDefaults
    if let userDefaults = UserDefaults(suiteName: "group.com.piripimartin.dew") {
      
      if let sharedData = userDefaults.object(forKey: "SharedData") as? [String: Any] {
        // Clear the data after reading
        userDefaults.removeObject(forKey: "SharedData")
        userDefaults.removeObject(forKey: "SharedDataTimestamp")
        userDefaults.synchronize()
        
        resolve(sharedData)
      } else {
        resolve(NSNull())
      }
    } else {
      reject("ERROR", "Could not access shared UserDefaults", nil)
    }
  }
}
